cli/azd/cmd/provision.go

```diff
@@ -166,16 +166,25 @@ func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 	if err != nil {
 		return nil, fmt.Errorf("creating provisioning manager: %w", err)
 	}
-
-	deploymentPlan, err := infraManager.Plan(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("planning deployment: %w", err)
-	}
+	var deployResult *provisioning.DeployResult
 
 	provisioningScope := infra.NewSubscriptionScope(
 		p.azCli, p.env.GetLocation(), p.env.GetSubscriptionId(), p.env.GetEnvName(),
 	)
-	deployResult, err := infraManager.Deploy(ctx, deploymentPlan, provisioningScope)
+	projectEventArgs := project.ProjectLifecycleEventArgs{
+		Project: p.projectConfig,
+	}
+
+	err = p.projectConfig.Invoke(ctx, project.ProjectEventProvision, projectEventArgs, func() error {
```

Comment:
Wrapping this block in `Invoke` ensures that the `preprovision` and `postprovision` internal hooks are fired so that container app service target can do its thing.